compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -143,6 +149,30 @@ void reportDiagnostic(String message, String diagnosticErrorCode, DiagnosticSeve
         this.diagnosticResult = new DefaultDiagnosticResult(this.diagnosticList);
     }
 
+    public boolean autoUpdate() {
+        return autoUpdate;
+    }
+
+    private boolean getSticky(PackageContext rootPackageContext) {
+        if (!rootPackageContext.project().buildOptions().sticky()) {
+            // set sticky if `build` file exists and `last_update_time` not passed 24 hours
+            Path buildFilePath = this.rootPackageContext.project().sourceRoot().resolve(TARGET_DIR_NAME)
```

Comment:
How does this work for bala projects since the targets in bala projects are tmp directories. I think we have a getTarget function in the project which we can use here